connectors/jdk-connector/src/main/java/org/glassfish/jersey/jdk/connector/internal/HttpConnection.java

```diff
@@ -127,7 +127,7 @@ public void completed(HttpRequest result) {
         });
     }
 
-    synchronized void close() {
```

Comment:
`close` should remain synchronized to prevent running `filterChain.close();` multiple times.
The deadlock can be better solved by disallowing a change from State.CLOSED to other states

Reply:
Actually this is ok, it is already called from a synchronized blocks, except `processConnectionClosed`.